src/main/kotlin/react/bootstrap/components/button/Button.kt

```diff
@@ -0,0 +1,273 @@
+        }
+
+        props.sizes?.also {
+            btnClasses.add(
+                when (it) {
+                    Sizes.SM -> ClassNames.BTN_SM
+                    Sizes.LG -> ClassNames.BTN_LG
+                }
+            )
+        }
+
+        props.variant?.also {
+            btnClasses.add(it.className)
+        }
+
+        if (props.type !== null) {
+            when (val type = props.type) {
+                is Types.Button -> {
+                    button(
+                        formEncType = type.formEncType,
+                        formMethod = type.formMethod,
+                        type = type.type,
+                        classes = props.className.appendClass(btnClasses)
+                    ) {
+                        attrs {
+                            handleAttrs()
+                            if (props.disabled == true) {
+                                attrs.disabled = true
+                            }
+                        }
+                        children()
+                    }
+                }
+                is Types.Input -> {
+                    input(
+                        type = type.type.inputType,
+                        formEncType = type.formEncType,
+                        formMethod = type.formMethod,
+                        name = type.name,
+                        classes = props.className.appendClass(btnClasses)
+                    ) {
+                        attrs {
+                            handleAttrs()
+                            defaultValue = type.value
+
+                            if (props.disabled == true) {
+                                disabled = true
+                            }
+                        }
+                        // No children allowed
+                    }
+                }
+                is Types.Link -> {
+                    a(
+                        href = type.href,
+                        target = type.target,
+                        classes = props.className.appendClass(btnClasses)
+                    ) {
+                        attrs {
+                            handleAttrs()
+                            role = "button"
+                        }
+                        children()
+                    }
+                }
+            }
+        } else {
+            button(
+                type = ButtonType.button,
+                classes = props.className.appendClass(btnClasses)
+            ) {
+                attrs {
+                    handleAttrs()
+                    if (props.disabled == true) {
+                        attrs.disabled = true
+                    }
+                }
+                children()
+            }
+        }
+    }
+
+    private fun <T : CommonAttributeGroupFacade> T.handleAttrs() {
+        if (props.active == true) {
+            ariaPressed = true
+        }
+
+        if (props.disabled == true) {
+            if (props.type is Types.Link) {
```

Comment:
move this into the when case for `Types.Link` or move all the property setting from when into `handleAttrs`